<launch>
  <!-- Disable if running online and not from a bag -->
  <param name="use_sim_time" value="false"/>

  <!-- Arguments -->
  <arg name="map_file" default="$(find my_amcl)/maps/sim.yaml"/>
  <arg name="open_rviz" default="true"/>

  <!-- Adding the my_map tf to the tf tree-->
  <node pkg="tf2_ros" type="static_transform_publisher" name="my_map_to_map" args="0 0 0 0 0 0 1 /disc_map /map "/>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find my_amcl)/rviz/test_mp3.rviz"/>
  </group>

  <!-- discretize map -->
  <node name="discretize_map" pkg="my_amcl" type="map_discretization.py" output="screen"/>

  <!-- Map saver -->
  <node name="disc_map_saver" pkg="map_server" type="map_saver" args="-f $(find my_amcl)/maps/disc_map">
      <remap from="/map" to="/disc_map"/>
  </node>

</launch>